dot/state/storage.go

```diff
@@ -270,11 +260,15 @@ func (s *StorageState) StorageRoot() (common.Hash, error) {
 	s.lock.RLock()
 	defer s.lock.RUnlock()
 
-	if s.tries[sr] == nil {
-		return common.Hash{}, errTrieDoesNotExist(sr)
+	t, ok := s.tries[sr]
+	if !ok {
+		t, err = s.LoadFromDB(sr)
+		if err != nil {
+			return common.Hash{}, errTrieDoesNotExist(sr)
+		}
 	}
 
-	return s.tries[sr].Hash()
+	return t.Hash()
```

Comment:
this function can be updated to just `return s.blockState.BestBlockStateRoot()` since it should be returning the current state root

Reply:
Done